linode/linode-blockstorage-csi-driver
# [Testing] Add New E2E Test Case & Enhance E2E Testing Configuration

## Reason for this PR:
New E2E test (pod-pvc-luks-mov-volume) being added to make sure we check for this edge case: https://github.com/linode/linode-blockstorage-csi-driver/issues/224

- Added E2E_SELECTOR environment variable to CI workflow for dynamic test selection based on branch.
- Updated e2e-test command to utilize E2E_SELECTOR for improved test execution control.
- Added labels ('all' and 'quick') to multiple test definitions for better categorization and selection.
- Renamed 'pod-pvc-expand-storage' test to 'pod-pvc-expand-storage-size' for clarity.

## How to run the new test and verify:
1. Create a 1 control plane and 1 worker node cluster with CSI driver deployed
    `CONTROLPLANE_NODES=1 WORKER_NODES=1 devbox run capl-cluster`
2. Make sure you are running CSI driver image with tag `statefulset-encrypt-pv-fix`. You can edit the resource then rollout restart. Link to image: https://hub.docker.com/layers/linode/linode-blockstorage-csi-driver/statefulset-encrypt-pv-fix/images/sha256-0da0820bbaf72a8becd5ba4cd998ba8f04e95a682415fdd8cee9a9dfa504dbf8?context=explore
3. Run the e2e-test command:
    `E2E_SELECTOR=volumemove CONTROLPLANE_NODES=1 WORKER_NODES=1 devbox run e2e-test`

For more info on e2e test setup, please refer: https://github.com/linode/linode-blockstorage-csi-driver/tree/main/tests/e2e

### General:


### Pull Request Guidelines:

1. [ ] Does your submission pass tests?
1. [ ] Have you added tests? 
1. [ ] Are you addressing a single feature in this PR? 
1. [ ] Are your commits atomic, addressing one change per commit?
1. [ ] Are you following the conventions of the language? 
1. [ ] Have you saved your large formatting changes for a different PR, so we can focus on your work?
1. [ ] Have you explained your rationale for why this feature is needed? 
1. [ ] Have you linked your PR to an [open issue](https://blog.github.com/2013-05-14-closing-issues-via-pull-requests/)
